app.js

```diff
@@ -12,10 +16,27 @@ const chatRoute = require('./server/routes/chat');
 app.use(cors({
     origin: '*'
 }));
+app.use(bodyParser.urlencoded({ extended: false }))
+app.use(bodyParser.json())
+app.use(morgan('dev'))
+
 app.use('/user', userRoute);
 app.use('/chat', userRoute);
 app.use('/setting', userRoute);
 
+app.use((req, res, next) => {
```

Comment:
Make a exception class for this 

Like this 

const ErrorResponse = require("../utils/errorResponse");

const errorHandler = (error, req, res, next) => { 
    let e = { ...error };
    e.message = error.message;

    // Log to console for dev
    console.log(error);
    // console.dir(error)

    // Moongoose Error
    if (error.name === "CastError") {
        const message = `Resources not found`;
        e = new ErrorResponse(message, 404)
    }

    if (error.code === 11000) {
        const message = `Duplicate field value entered`;
        e = new ErrorResponse(message, 400)
    }

    if (error.name === "ValidationError") {
        const message = Object.values(error.errors).map(val => val.message);
        e = new ErrorResponse(message, 400)

        
    }
    // console.log(error.name);

    res.status(e.statusCode || 500).json({
        success: false,
        error: e.message || 'Server Error'
    });
}

module.exports = errorHandler;

class ErrorResponse extends Error { 
    constructor(message, statusCode) { 
        super(message);
        this.statusCode = statusCode;

    }
}

module.exports = ErrorResponse;

And use it like this 

if (!email || !password) {

        return next(new ErrorResponse("Please provide an email and password", 400));

    }